db/rdb.go

```diff
@@ -342,39 +342,33 @@ func (r *RDBDriver) GetByCpeURI(uri string) ([]models.CveDetail, error) {
 	return details, nil
 }
 
+// CountJvn count jvn table
+func (r *RDBDriver) CountJvn() (int, error) {
+	var count int64
+	err := r.conn.Model(&models.Jvn{}).Count(&count).Error
+	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+		return 0, err
+	}
+	return int(count), nil
+}
+
 // InsertJvn inserts Cve Information into DB
 func (r *RDBDriver) InsertJvn(feedMetas []models.FeedMeta, cves map[string][]models.Jvn) error {
-	logger.Infof("Inserting fetched CVEs...")
-	for _, meta := range feedMetas {
-		result := r.conn.Where(&models.FeedMeta{Source: meta.Source, Year: meta.Year}).Take(&meta)
-		if result.Error != nil && !errors.Is(result.Error, gorm.ErrRecordNotFound) {
-			return fmt.Errorf("Failed to select. err: %s", result.Error)
-		}
-
-		if result.RowsAffected == 0 {
-			if err := r.conn.Create(&meta).Error; err != nil {
-				return xerrors.Errorf("Failed to insert. err: %w", err)
-			}
-		}
-
-		if _, ok := cves[meta.Year]; !ok {
-			return xerrors.Errorf("Failed to get cves[%s].", meta.Year)
-		}
-
-		if err := r.deleteAndInsertJvn(r.conn, meta.ID, cves[meta.Year]); err != nil {
+	for y, cve := range cves {
+		logger.Infof("Inserting fetched CVEs(%s)...", y)
+		if err := r.deleteAndInsertJvn(r.conn, cve); err != nil {
 			return xerrors.Errorf("Failed to deleteAndInsertJvn. err: %w", err)
 		}
+	}
 
-		meta.Hash = meta.LatestHash
-		meta.LastModifiedDate = meta.LatestLastModifiedDate
-		if err := r.conn.Save(&meta).Error; err != nil {
-			return xerrors.Errorf("Failed to insert. err: %w", err)
-		}
+	logger.Infof("Updating FeedMetas...")
+	if err := r.upsertFeedMetas(feedMetas); err != nil {
+		return xerrors.Errorf("Failed to upsertFeedMetas. err: %w", err)
 	}
 	return nil
 }
 
-func (r *RDBDriver) deleteAndInsertJvn(conn *gorm.DB, feedMetaID uint, cves []models.Jvn) (err error) {
+func (r *RDBDriver) deleteAndInsertJvn(conn *gorm.DB, cves []models.Jvn) (err error) {
```

Comment:
Is there any reason not to use r.Conn?

Reply:
I don't know why, but I think they were wondering whether to implement it as an RDBDriver Method or a function.